smoke-tests/apps/PreAggMetricsWithRoleNameOverridesAndSampling/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/PreAggMetricsWithRoleNameOverridesAndSamplingTest.java

```diff
@@ -0,0 +1,228 @@
+import static com.microsoft.applicationinsights.smoketest.WarEnvironmentValue.TOMCAT_8_JAVA_8;
+import static com.microsoft.applicationinsights.smoketest.WarEnvironmentValue.TOMCAT_8_JAVA_8_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.WarEnvironmentValue.WILDFLY_13_JAVA_8;
+import static com.microsoft.applicationinsights.smoketest.WarEnvironmentValue.WILDFLY_13_JAVA_8_OPENJ9;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.microsoft.applicationinsights.smoketest.schemav2.Data;
+import com.microsoft.applicationinsights.smoketest.schemav2.DataPoint;
+import com.microsoft.applicationinsights.smoketest.schemav2.Envelope;
+import com.microsoft.applicationinsights.smoketest.schemav2.MessageData;
+import com.microsoft.applicationinsights.smoketest.schemav2.MetricData;
+import com.microsoft.applicationinsights.smoketest.schemav2.RemoteDependencyData;
+import com.microsoft.applicationinsights.smoketest.schemav2.RequestData;
+import com.microsoft.applicationinsights.smoketest.schemav2.SeverityLevel;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@UseAgent
+abstract class PreAggMetricsWithRoleNameOverridesAndSamplingTest {
+
+  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
+
+  private static final int COUNT = 100;
+  private static final String DEFAULT_ROLE_NAME = "testrolename";
+
+  @Test
+  @TargetUri(value = "/app2", callCount = COUNT)
+  void testApp2() throws Exception {
+    testApp("app2");
+  }
+
+  @Test
+  @TargetUri(value = "/app3", callCount = COUNT)
+  void testApp3() throws Exception {
+    testApp("app3");
+  }
+
+  private void testApp(String roleName) throws Exception {
+    // super super low chance that number of sampled requests is less than 25
+    long start = System.nanoTime();
+    while (testing.mockedIngestion.getCountForType("RequestData") < 25
+        && NANOSECONDS.toSeconds(System.nanoTime() - start) < 10) {}
+    // wait ten more seconds before checking that we didn't receive too many
+    Thread.sleep(SECONDS.toMillis(10));
+
+    List<Envelope> requestEnvelopes =
+        testing.mockedIngestion.getItemsEnvelopeDataType("RequestData");
+    Thread.sleep(SECONDS.toMillis(10));
+
+    List<Envelope> metricsEnvelops = testing.mockedIngestion.getItemsEnvelopeDataType("MetricData");
+    // TODO need to fix pre agg metrics for non-sampled spans (PropagatedMetric) with role name
+    // overrides
+    List<Envelope> clientRoleNameOverriddenEnvelops = new ArrayList<>();
+    List<Envelope> clientRoleNameNotOverriddenEnvelops = new ArrayList<>();
+    List<Envelope> serverRoleNameOverriddenEnvelops = new ArrayList<>();
+    List<Envelope> serverRoleNameNotOverriddenEnvelops = new ArrayList<>();
```

Comment:
need to update this when testing your pr @trask. here is 50% rolenames are updated.

Reply:
you can merge yours first and then i can update here.